src/components/utrecht/Select.test.tsx

```diff
@@ -0,0 +1,408 @@
+
+    expect(ref.current).toBe(select);
+  });
+});
+
+describe("Select option", () => {
+  it("renders an option role element", () => {
+    render(<SelectOption />);
+
+    const option = screen.getByRole("option");
+
+    expect(option).toBeInTheDocument();
+  });
+
+  it("renders an option role element with a text label", () => {
+    render(<SelectOption>Heading 1</SelectOption>);
+
+    const option = screen.getByRole("option", {
+      name: "Heading 1",
+    });
+
+    expect(option).toBeInTheDocument();
+  });
+
+  it("renders an HTML option element", () => {
+    const { container } = render(<SelectOption />);
+
+    const option = container.querySelector("option:only-child");
+
+    expect(option).toBeInTheDocument();
+  });
+
+  it("renders a design system BEM block class name", () => {
+    const { container } = render(<SelectOption />);
+
+    const select = container.querySelector(":only-child");
+
+    expect(select).toHaveClass("utrecht-select__option");
+  });
+
+  // Rich text is not supported by the HTML <option> element
+
+  describe("disabled variant", () => {
+    it("can have a disabled state", () => {
+      const { container } = render(<SelectOption disabled />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).toBeDisabled();
+    });
+
+    it("is not disabled by default", () => {
+      const { container } = render(<SelectOption />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).not.toBeDisabled();
+    });
+
+    it("renders a design system BEM modifier class name", () => {
+      const { container } = render(<SelectOption disabled />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).toHaveClass("utrecht-select__option--disabled");
+    });
+
+    it("omits non-essential disabled attributes when not disabled", () => {
+      const { container } = render(<SelectOption disabled={false} />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).not.toHaveAttribute("aria-disabled");
+
+      expect(select).not.toHaveAttribute("disabled");
+    });
+
+    it("can have a disabled state in CSS", () => {
+      const { container } = render(<SelectOption disabled />);
+
+      const select = container.querySelector(":disabled");
+
+      expect(select).toBeInTheDocument();
+    });
+  });
+
+  describe("invalid variant", () => {
+    it("can have a invalid state", () => {
+      const { container } = render(<SelectOption invalid />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).toBeInTheDocument();
+    });
+
+    it("is not invalid by default", () => {
+      const { container } = render(<SelectOption />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).not.toBeInvalid();
+      expect(select).not.toHaveClass("utrecht-select__option--invalid");
```

Comment:
I like this way of checking a modifier class is not present if the invalid attribute is not set. This is what I would also like for the disabled etc. properties of the Select component above